packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts

```diff
@@ -52,6 +52,10 @@ export class StepFunctionsStartExecution extends sfn.TaskStateBase {
   constructor(scope: Construct, id: string, private readonly props: StepFunctionsStartExecutionProps) {
     super(scope, id, props);
 
+    if (props.input && !(props.input instanceof sfn.TaskInput)) {
+      throw new Error('StepFunctionsStartExecution `input` must be of type `TaskInput`.');
+    }
```

Comment:
i believe we use TaskInput in other locations without requiring this check.
Is it possible that the issue is in rendering the content itself? (an example of where we use the TaskInput as a parameter would be `LambdaInvoke`)

Reply:
Same problem arises in `LambdaInvoke` - it appears this is a wider problem with `TaskInput`.